drain.go

```diff
@@ -279,5 +325,74 @@ func (s *server) serveDrain(w http.ResponseWriter, r *http.Request) {
 
 	parseTimer.UpdateSince(parseStart)
 
+	if len(shadowURLs) > 0 {
+		go amplify(r.Header, buf)
+	}
 	w.WriteHeader(http.StatusNoContent)
 }
+
+func (s *server) serveShadowURLs(w http.ResponseWriter, r *http.Request) {
+	if r.Method != "POST" {
+		w.WriteHeader(http.StatusMethodNotAllowed)
+		return
+	}
+
+	var urls []string
+	decoder := json.NewDecoder(r.Body)
+	if err := decoder.Decode(&urls); err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	if err := setShadowURLs(urls); err != nil {
+		log.Println(err)
+	}
+	w.WriteHeader(http.StatusNoContent)
+}
+
+func getShadowURLs() map[string]*big.Int {
```

Comment:
I'd expect to be able to introspect URLs so we could more easily tack on another URL.